src/backend/linux_raw/io/epoll.rs

```diff
@@ -229,84 +211,68 @@ impl<Context: self::Context> Epoll<Context> {
                 timeout,
             )?;
             event_list.events.set_len(nfds);
-            event_list.context = &self.context;
         }
 
         Ok(())
     }
 }
 
 #[cfg(feature = "std")]
-impl<'context, T: AsFd + Into<OwnedFd> + From<OwnedFd>> AsRawFd for Epoll<Owning<'context, T>> {
+impl AsRawFd for Epoll {
     fn as_raw_fd(&self) -> RawFd {
         self.epoll_fd.as_raw_fd()
     }
 }
 
 #[cfg(feature = "std")]
-impl<'context, T: AsFd + Into<OwnedFd> + From<OwnedFd>> IntoRawFd for Epoll<Owning<'context, T>> {
+impl IntoRawFd for Epoll {
     fn into_raw_fd(self) -> RawFd {
         self.epoll_fd.into_raw_fd()
     }
 }
 
 #[cfg(feature = "std")]
-impl<'context, T: AsFd + Into<OwnedFd> + From<OwnedFd>> FromRawFd for Epoll<Owning<'context, T>> {
+impl FromRawFd for Epoll {
     unsafe fn from_raw_fd(fd: RawFd) -> Self {
         Self {
             epoll_fd: OwnedFd::from_raw_fd(fd),
-            context: Owning::new(),
         }
     }
 }
 
 #[cfg(feature = "std")]
-impl<'context, T: AsFd + Into<OwnedFd> + From<OwnedFd>> AsFd for Epoll<Owning<'context, T>> {
+impl AsFd for Epoll {
     fn as_fd(&self) -> BorrowedFd<'_> {
         self.epoll_fd.as_fd()
     }
 }
 
 #[cfg(feature = "std")]
-impl<'context, T: AsFd + Into<OwnedFd> + From<OwnedFd>> From<Epoll<Owning<'context, T>>>
-    for OwnedFd
-{
-    fn from(epoll: Epoll<Owning<'context, T>>) -> Self {
+impl From<Epoll> for OwnedFd {
+    fn from(epoll: Epoll) -> Self {
         epoll.epoll_fd
     }
 }
 
 #[cfg(feature = "std")]
-impl<'context, T: AsFd + Into<OwnedFd> + From<OwnedFd>> From<OwnedFd>
-    for Epoll<Owning<'context, T>>
-{
+impl From<OwnedFd> for Epoll {
     fn from(fd: OwnedFd) -> Self {
-        Self {
-            epoll_fd: fd,
-            context: Owning::new(),
-        }
+        Self { epoll_fd: fd }
     }
 }
 
 /// An iterator over the `Event`s in an `EventVec`.
-pub struct Iter<'context, Context: self::Context> {
-    iter: core::slice::Iter<'context, Event>,
-    context: *const Context,
-    _phantom: PhantomData<&'context Context>,
+pub struct Iter<'a> {
+    iter: core::slice::Iter<'a, Event>,
 }
 
-impl<'context, Context: self::Context> Iterator for Iter<'context, Context> {
-    type Item = (EventFlags, Ref<'context, Context::Target>);
+impl<'a> Iterator for Iter<'a> {
+    type Item = (EventFlags, u64);
```

Comment:
If we're going to hand the raw `u64` back to the user, we should rename the `encoded` field to just `data`, and drop the `encode`/`decode` functions in the `Context` abstraction.